src/Translation/Translator.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Translation;
+
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\System;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Translator.
+ *
+ * @author Martin Auswöger <martin@auswoeger.com>
+ */
+class Translator implements TranslatorInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    /**
+     * @param TranslatorInterface      $translator Original translator service that gets decorated
+     * @param ContaoFrameworkInterface $framework
+     */
+    public function __construct(TranslatorInterface $translator, ContaoFrameworkInterface $framework)
+    {
+        $this->translator = $translator;
+        $this->framework = $framework;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function trans($id, array $parameters = [], $domain = null, $locale = null)
```

Comment:
the symfony interface is php5, it is ok to use parameter hints in the implementing method, also return type declaration is missing

Reply:
same for following methods

Reply:
No, parameter hints are not allowed for implemented methods in PHP 7.1, only in 7.2

Reply:
you are absolutely right on this. Thought I had done this successfully in a project but was wrong